src/components/Layout.tsx

```diff
@@ -12,11 +12,36 @@ import "./Layout.css"
 
 import type { SortOption } from "./Store"
 
+const handleSubmit = (e, setState) => {
+  e.preventDefault()
+  try {
+    JSON.parse(e.target.firstChild.value)
+  } catch (e) {
+    console.log("Invalid JSON string")
+  }
+  localStorage.setItem('filter-rules', e.target.firstChild.value)
+  setState(e.target.firstChild.value)
+}
+
+const Rules = (props) => (
+  <div id="match-rules-page" className="match-rules">
+    <label htmlFor="match-rules">
+      Filter rules
+    </label>
+    <form onSubmit={e => handleSubmit(e, props.setState)}>
+      <textarea id="match-rules" name="match-rules-area" rows="40" cols="50" defaultValue={props.state} />
+      <button type="submit">Save</button>
+    </form>
+  </div>
+) 
```

Comment:
I would probaby simplify this to something like
```suggestion
const Rules = (props) => (
  const handleChange = (e) => {
    e.preventDefault()
    let value = e.target.value
    try {
      JSON.parse(value)
    } catch (e) {
      warn("Invalid JSON string")
    }
    localStorage.setItem('filter-rules', value)
    setState(value)
  }

  <div id="match-rules-page" className="match-rules">
    <label htmlFor="match-rules">
      Filter rules
    </label>
      <textarea id="match-rules" name="match-rules-area" rows="40" cols="50" onChange={handleChange} value={props.state} />
  </div>
) 
```

Reply:
Does look cleaner (though I already added more options anyways), but didn't work out of the box. Version that kinda worked ran into an issue I already fought against earlier today where each time you type a letter in the textbox it loses focus, making it unusable.

Reply:
aah 👍